### PR TITLE
Reading files which are open by other process.

### PR DESCRIPTION
**Pull request by seba76**
Currently driver can't read files which are open by other processes, proposed changes in this pull request enable this.

### CODE REVIEW

**Comment by i2van** [comment]
Looks good.

`FileStream` uses `new FileStream(path, FileMode.Open, FileAccess.Read, FileShare.Read)` internally, so I think I'll add an option for using open files read. What do you think?

File: Src/CsvLINQPadDriver/Extensions/FileExtensions.cs [hunk 1]
```diff
@@ -536,7 +537,7 @@ Encoding GetEncoding()
             {
                 Charset charset;
 
-                using (var stream = File.OpenRead(fileName))
+                using (var stream = new FileStream(fileName, FileMode.Open, FileAccess.Read, FileShare.ReadWrite))
```
Comment (i2van) [inline Src/CsvLINQPadDriver/Extensions/FileExtensions.cs:540]: `FileShare.ReadWrite` means that other processes can read and write to a file opened by this process. It doesn't solve locked file issue, i.e. if another process opens file without specifying `FileShare.Read` nothing can be done about it.

`File.OpenRead` currently used internally creates file stream with `FileShare.Read`: `new FileStream(path, FileMode.Open, FileAccess.Read, FileShare.Read)`

Comment (seba76) [reply Src/CsvLINQPadDriver/Extensions/FileExtensions.cs:540]: I use NLog for writing tab separated files and unfortunately driver was not been able to open such files, I have daily rollover so current file can't be read. My idea was to add that possibility so I can perform queries in such files, I'm fine with the option as long as I can open such file.

Comment (seba76) [reply Src/CsvLINQPadDriver/Extensions/FileExtensions.cs:540]: Of course if file is locked by application there is no help, but opening files like proposed helps when reading logger files.

Comment (i2van) [reply Src/CsvLINQPadDriver/Extensions/FileExtensions.cs:540]: Implemented in [6.20.0](https://www.nuget.org/packages/CsvLINQPadDriver/6.20.0).

Check `Do not lock files being read option` to get expected behavior.

**Pull request merged**